STATIC PREDICTION TABLES.md

```diff
@@ -0,0 +1,96 @@
+# About this document
+
+This is a Work in Progress.
+
+This document describes a format for storing Static Prediction Tables.
+
+It is possible that some of all of this format will be usable as part of
+individual compressed source files, to be determined.
+
+# About Static Prediction Tables (SPT)
+
+A Static Prediction Table is a form of external dictionary, shipped either
+with the JavaScript VM, or separately, and which may be referenced by any
+number of compressed Binary AST Source Files.
+
+Shipping Static Prediction Tables makes it possible to considerably reduce
+the size of individual compressed files.
+
+This document does not attempt to document how and when a SPT is loaded,
+or how and when an individual compressed file references a SPT.
+
+# Design guidelines
+
+- A SPT must be usable by many compressed source files.
+- A VM must be able to manage several SPTs simultaneously.
+- As JavaScript is a changing language, a SPT that is complete at a given point in time may not be expected to remain
+  complete forever.
+- As Binary AST never reuses the same interface name for distinct purposes, a Path in the AST that is valid at a point
+  in time will remain valid forever.
+- For upgrade purposes, a SPT may be defined as an amendment to another SPT.
+- A SPT may define additional strings of various natures.
+
+# Format
+
+## Header
+
+The header:
+- specifies the kind of file;
+- references the grammar version;
+- optionally, references a SPT file it amends.
```

Comment:
At this early stage, "delta" SPTs do not seem valuable to spend effort in speccing.  They are off the fastpath anyway, and I can see them adding a lot of complexity to the spec.  Let's leave the deltas until we actually feel we need them.

Reply:
It's not the highest priority, but let's keep an eye on the road :)